src/main/java/org/truffleruby/debug/TruffleDebugNodes.java

```diff
@@ -460,4 +460,44 @@ public Object foreignObject() {
 
     }
 
+    @CoreMethod(names = "thread_info", onSingleton = true)
+    public abstract static class ThreadInfoNode extends CoreMethodArrayArgumentsNode {
+
+        @Child private StringNodes.MakeStringNode makeStringNode = StringNodes.MakeStringNode.create();
+
+        @Specialization
+        public DynamicObject threadInfo() {
+            return makeStringNode.executeMake(getThreadDebugInfo(), UTF8Encoding.INSTANCE, CodeRange.CR_UNKNOWN);
```

Comment:
I'd be surprised if this is anything but `CodeRange.CR_7BIT`, but it probably doesn't matter much. You'll just do a potentially unnecessary byte scan if `CR_UNKNOWN`.